File: SwiftAssignments/SwiftAssignments/Shared/Question.swift
import Foundation

protocol Question {
    func question1()
    func question2()
    func question3()
    func question4()
}

extension Question {
    func execute(number: Int) {
        switch number {
        case 1:
            question1()
        case 2:
            question2()
        case 3:
            question3()
        case 4:
            question4()
        default:
            break
        }
    }
}

File: RubyRewrite/lib/shared/question.rb
class Question
  def question1; end
  def question2; end
  def question3; end
  def question4; end

  def execute(number)
    case number
    when 1
      question1
    when 2
      question2
    when 3
      question3
    when 4
      question4
    end
  end
end
